src/main/java/org/jenkinsci/plugins/workflow/log/DelayBufferedOutputStream.java

```diff
@@ -54,6 +58,45 @@ private Tuning() {}
         static final Tuning DEFAULT = new Tuning();
     }
 
+    /**
+     * The interesting state of the buffered stream, kept as a separate object so that {@link FlushRef} can hold on to it.
+     */
+    private static final class Buffer {
```

Comment:
Why not just use a [NIO buffer](https://docs.oracle.com/javase/7/docs/api/java/nio/Buffer.html)?

Reply:
(In that case we'd store the stream separately from a Buffer. )

Reply:
I see no advantage in doing so.